foreign_cc/private/framework.bzl

```
@@ -893,25 +893,43 @@ def _expand_command_path(binary, path, command):
     else:
         return command
 
-def expand_locations(ctx, environ, data):
+def expand_locations(ctx, expandable, data):
     """Expand locations on a dictionary while ensuring `execpath` is always set to an absolute path
 
     This function is not expected to be passed to any action.env argument but instead rendered into
     build scripts.
 
     Args:
         ctx (ctx): The rule's context object
-        environ (dict): A dictionary of environment variables
+        expandable (dict, list, str): Variables to expand, can be a variety of different types
         data (list): A list of targets
 
     Returns:
         dict: An expanded dict of environment variables
     """
-    expanded_env = dict()
-    for key, value in environ.items():
+    if type(expandable) == type(dict()):
+        expanded_env = dict()
+        for key, value in expandable.items():
+            # If `EXT_BUILD_ROOT` exists in the string, we assume the user has added it themselves
+            if "EXT_BUILD_ROOT" in value:
+                expanded_env.update({key: ctx.expand_location(value, data)})
+            else:
+                expanded_env.update({key: ctx.expand_location(value.replace("$(execpath ", "$EXT_BUILD_ROOT/$(execpath "), data)})
```

Should we use the `$$VAR$$` form of env variables here? I can't remember the order in which these things get expanded.
```suggestion
                expanded_env.update({key: ctx.expand_location(value.replace("$(execpath ", "$$EXT_BUILD_ROOT$$/$(execpath "), data)})
```